packages/grid/x-data-grid/src/hooks/features/rows/gridRowsUtils.ts

```diff
@@ -389,5 +389,5 @@ export function getMinimalContentHeight(
   rowHeight: number,
 ) {
   const densityFactor = gridDensityFactorSelector(apiRef);
-  return 2 * Math.floor(rowHeight * densityFactor);
+  return `var(--DataGrid-noRowsOverlayHeight, ${2 * Math.floor(rowHeight * densityFactor)}px)`;
```

Comment:
I feel it doesn't strictly relate to the `NoRowsOverlay` rather it applies in general to all the overlays, for instance in this [demo](https://codesandbox.io/s/great-bird-nv953p), I tried to set a height of `500px` and show a loading overlay which also reflects the same height.
```suggestion
  return `var(--DataGrid-overlayHeight, ${2 * Math.floor(rowHeight * densityFactor)}px)`;
```

Reply:
Huh, this was not my intention, but it actually makes sense! 😅
Thanks for spotting it!